src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -38,32 +44,78 @@
             + "2. " + COMMAND_WORD + " 1";
 
     public static final String MESSAGE_UNKNOWN_PERSON_NAME = "The person named %s is not found.";
-
+    public static final String MESSAGE_UNKNOWN_PERSON_TAG = "Persons with tag %s are not found.";
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_TAG_SUCCESS = "Deleted Contacts Under Tag: %1$s";
 
     private final Index targetIndex;
     private final Name targetName;
+    private final Predicate<Person> targetPerson;
```

Comment:
The naming for Predicate<Person> variable is not very clear to me but I am not sure what is the better name for it :/